### ISSUE TITLE
[✨] Update the Qwik Library starter

### ISSUE THREAD

**Issue by shairez** [BuilderIO/qwik#5559]
### Is your feature request related to a problem?

Following up on #5097 

### Describe the solution you'd like

We need to update the starter `vite.config.ts` and add 

``` 
dts({
	// ...
	entryRoot: 'src'
})
```

and also 
```
build: {  
  // ...
  rollupOptions: {
    output: {
        preserveModules: true,
        preserveModulesRoot: 'packages/kit-headless/src',
      },
  }
}  
```

### Describe alternatives you've considered

not relevant

### Additional context

I think we should implement this fix only after we fix https://github.com/BuilderIO/qwik/issues/5473

**Comment by maiieul** [comment]
I agree, we should wait until #5473 is fixed before doing this. I'd currently rather have the "big" bundle rather than the namespace issues `preserveModules` creates.

**Comment by gioboa** [comment]
Is it still valid?

**Comment by maiieul** [comment]
Yes but I think this has been implemented a while ago 😅

**Comment by gioboa** [comment]
I close this because we already have preserveModules in the library vite.config and 
```
dts({
  // ...
  entryRoot: 'src'
})
``` 
is something related to Nx and we don't have the specific starter for this.
